wiring/inc/spark_wiring_ticks.h

```diff
@@ -26,13 +26,18 @@
 #ifndef SPARK_WIRING_TICKS_H
 #define	SPARK_WIRING_TICKS_H
 
+#include "timer_hal.h"
```

Comment:
I want to specifically call this out. `timer_hal.h` is order dependent. It cannot come after `algorithm`. To accommodate this, I needed to move `timer_hal.h` outside of the `extern "C" {` block.

In doing so, I also pulled `delay_hal.h` outside of the `extern "C" {` block, because it looks like it already appropriately handles `extern "C" {` on it's API.

Reply:
This is most likely an artifact of including a header within an `extern "C"` block and we should not be doing so.

The only use-case for this is if we are including a third-party C-only header that we can't control/modify/fork in C++ code that does not mark the functions with `extern "C"` when included from C++ code.